tls-client/main.cpp

```diff
@@ -132,6 +133,8 @@ class HelloHTTPS {
         mbedtls_x509_crt_free(&_cacert);
         mbedtls_ssl_free(&_ssl);
         mbedtls_ssl_config_free(&_ssl_conf);
+        _tcpsocket->close();
```

Comment:
I see more lines of code to manage closing the socket in ```startTest()``` than putting it in the constructor. We have to see trade off between coherence and pollution in code. For coherence  we should also return from a single point in ```startTest```.